driver/modern_bpf/definitions/struct_flavors.h

```diff
@@ -36,4 +36,21 @@ struct modern_bpf__kernel_timex_timeval
 	long long int tv_usec;
 };
 
+/* COS kernels handle audit field differently, see [1]. To support both
+ * versions define COS subset of task_struct with a flavor suffix (which will
+ * be ignored during relocation matching [2]).
+ *
+ * [1]: https://chromium.googlesource.com/chromiumos/third_party/kernel/+/096925a44076ba5c52faa84d255a847130ff341e%5E%21/#F2
+ * [2]: https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/tools/lib/bpf/libbpf.c#n5347
+ */
+struct audit_task_info {
```

Comment:
you have to move them inside `#pragma clang attribute` blocks otherwise no relocation will be performed

```c
#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
#pragma clang attribute push(__attribute__((preserve_access_index)), apply_to = record)
#endif

... // <-- here

#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
#pragma clang attribute pop
#endif
```

Other structures with `modern_bpf__` prefix are defined by us so they don't need relocation, but this is not the case

Reply:
Oh, haven't noticed that. Done.